include/glow/Importer/ProtobufLoader.h

```diff
@@ -105,12 +105,32 @@ class ProtobufLoader {
 
 public:
   /// \returns the node that was registered with the name \p name.
-  Node *getNodeByName(const std::string &name);
+  /// When no node has been registered for this name, the method will
+  /// either return nullptr if \p allowNullptr is true or will crash the
+  /// program otherwise. In other words, this can be used to perform
+  /// invariant checks when we know something must not be nullptr.
+  Node *getNodeByName(const std::string &name, bool allowNullptr = false) const;
 
   /// \returns the node that was registered with the name \p name or create a
   /// new Variable node for a tensor with this name.
   Node *getOrCreateNodeByName(const std::string &name);
 
+  /// Create a new node \p name initialized with \p tensor.
+  /// \returns the newly created node.
+  Node *
+  createNode(const std::string &name, Tensor &tensor,
+             VisibilityKind visibilityKind = VisibilityKind::Private,
```

Comment:
nit: I'm a bit unsure about having Variable's defaults copied here. In case we change it there, we'd have to change here also. But I don't have better suggestion. In case we don't give these defaults, having to write VisibilityKind etc everywhere is also annoying. What do you think?

Reply:
I was thinking the same and I don't have a good suggestion either.
I've tried to mitigate the problem by only exposing these parameters to the createAndRememberNode method and making it protected. That should limit its users.